docs/content/template-authoring.md

```diff
@@ -0,0 +1,86 @@
+title = "Creating Spin templates"
+template = "main"
+date = "2022-03-14T00:22:56Z"
+[extra]
+url = "https://github.com/fermyon/spin/blob/main/docs/content/template-authoring.md"
+---
+
+Spin templates allow a Spin developer to quickly create the skeleton of an
+application or component, ready for the application logic to be filled in.
+
+A template consists of two directories, `content` and `metadata`.
```

Comment:
Somewhat :bike::derelict_house:y: what is the rationale for having both of these directories? It seems like one of them could be flattened into the root?

Reply:
Because it _guarantees_ a clean separation.  In future a template might have scripts, optional files, IDE icons, who knows what; by hiving off the execution stuff from the payload we allow arbitrary extension without any risk of a future clash.

That said, we could say that the manifest goes in the root and the payload in `content`, and when we add other things they can go in a `metadata` or other more specifically named directory at that point.  It doesn't really much matter.  But I don't want to end up with a grab bag of stuff in the root and the `content` kind of lost among it all.

Reply:
Having the manifest in the root feels more "familiar" to me, but I think its just an aesthetic preference.

Reply:
I am critically _weary _of the cost of trying to "gaurantee" this clean separation. 

1, it constrains the flexibility a template author has over how they organize their repository and code they maintain. 
2. The assumptions here is that `spin-template.toml` must exist in the parent of any directory named `content` and `metadata` to qualify as a template. This implies that as an author I am required to duplicate parameter definitions, licenses into each tempate's metadata directory on a  *per* template basis. 
3. As a template author, I want to frame the interactive experience for a consumer (done so through `spin-template.toml`) however, given the current design, I am forced to develop a unique (or duplicate) my interactive experience for each template i offer.

Reply:
I am heavily leaning towards the only assumption about a repository is it contains a (or many) `spin-template.toml` somewhere in the repo. Everything part of the template is relative to where the `spin-template.toml` is located. 

Basically, the way the original prototype worked